server/cert.go

```diff
@@ -0,0 +1,107 @@
+package server
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"math/big"
+	"net/url"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+func genCert(host string, priv *ecdsa.PrivateKey) ([]byte, error) {
+	serial, err := rand.Int(rand.Reader, new(big.Int).Lsh(big.NewInt(1), 128))
+	glog.Info("Generating cert for ", host)
+	if err != nil {
+		glog.Error("Could not generate serial ", err)
+		return []byte{}, err
+	}
+	tmpl := x509.Certificate{
+		SerialNumber:          serial,
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(8765 * time.Hour), // XXX fix fix fix
+		KeyUsage:              x509.KeyUsageCertSign,
```

Comment:
Testing seems to be OK without it, and in any case we ignore checking the signature for TLS, so...

Do the docs indicate that the field is required? I can't find anything saying so. From reviewing the implementation, it seems that a default value is chosen anyway based on the type of the cert. https://golang.org/src/crypto/x509/x509.go

Reply:
Was just asking because I didn't see anything in the docs and for the case where we do check the signature, but based upon the implementation `signingParamsForPublicKey()`, it seems like its not required so we are 👍